kamon-annotation/src/main/scala/kamon/annotation/util/ELProcessorPool.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2015 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.annotation.util
+
+import javax.el.ELProcessor
+
+import kamon.Kamon
+import kamon.annotation.Annotation
+import kamon.annotation.resolver.PrivateFieldELResolver
+import kamon.annotation.util.FastObjectPool.PoolFactory
+
+/**
+ * Convenient pool of @see ELProcessor, since it is not thread safe and the creation is expensive.
+ */
+object ELProcessorPool {
+  private val pool = new FastObjectPool[ELProcessor](ELPoolFactory(), Kamon(Annotation).poolSize)
```

Comment:
What about just having a ThreadLocal[ELProcessor]? Having a too conservative size for the object pool can generate unnecessary blocks in the code... I'm not entirely sure about introducing the object pool. what do you thing? what would be the pros/cons of using a simple ThreadLocal vs FastObjectPool?